infrastructure_source_code.md

```diff
@@ -0,0 +1,44 @@
+# DataWorks Open Source Infrastructure Source Code Policy
+
+## Purpose
+
+This document sets out local policy, for the Department for Work and Pensions (DWP) DataWorks team, on what types of information can and cannot be included in infrastructure code that is made publicly available.
+
+It is expected that, at some point, this document will be superceded by a DWP-wide open-source infrastructure code policy which encompasses all DWP public GitHub repositories. Some useful [DWP guidance on creating open source products](https://confluence.service.dwpcloud.uk/pages/viewpage.action?title=Guidance+on+creating+open+source+products&spaceKey=EN) currently exists, but it has not yet been made open-source.
+
+## Scope
+
+This policy covers infrastructure code. Please refer to the [index](README.md#index) for links to other related polices.
+
+## General Approach
+
+Infrastructure code should be made open wherever it is appropriate to do so:
+
+This approach makes code more readily accessible to a broad target audience, so that they can be more easily used for their intended purpose;
+1. This approach enables and encourages re-use of common standards across the DWP and other Government departments, which in turn makes collaboration easier. It also enables other parties to feed back improvements, which benefits everyone.
+1. This policy is in line with the spirit of the  [Government Digital Service (GDS) guidelines on making information open-source](https://gds-operations.github.io/guidelines/).
+
+## Security Restrictions
```

Comment:
We need to make a distinction between working on code and working with data. While we'd want to open-source infrastructure code development, and potentially open the door to a BYOD-type way of working for infra, we would most certainly not want to do this for any development work which would require data access. This distinction becomes more relevant for application code development, but may also be relevant for some infrastructure code work and so should be considered here.

Reply:
None of our work should require access to data. Certainly development work should never involve any data that isn't synthetic.

Reply:
Amen to that. However, working in other environments with Devs who were perhaps somewhat less professional than the current team this was always a constant argument (particularly where insecure ways of working were ingrained within the team before I arrived), particularly around the alleged grey areas of what constitutes anonymised data. If only to support your above point, we still need to make a distinction between working on code and working with data. The policy should be explicit about "no real data" and "no anonymised data" and that infra development work should never involve any data that isn't synthetic.